tidb: add a new system variable `tidb_service_scope` (#14796)
pingcap/docs: tidb-distributed-execution-framework.md

    SET GLOBAL tidb_enable_dist_task = ON;
    ```

    When backend tasks are running, the statements supported by the framework ( such as Add index and `IMPORT INTO`) are executed in a distributed manner. All TiDB nodes will run backend tasks by default.

2. Adjust the following system variables that might affect the distributed execution of DDL tasks according to your needs:

    * [`tidb_ddl_reorg_worker_cnt`](/system-variables.md#tidb_ddl_reorg_worker_cnt): use the default value `4`. The recommended maximum value is `16`.
    * [`tidb_ddl_reorg_priority`](/system-variables.md#tidb_ddl_reorg_priority)
    * [`tidb_ddl_error_count_limit`](/system-variables.md#tidb_ddl_error_count_limit)
    * [`tidb_ddl_reorg_batch_size`](/system-variables.md#tidb_ddl_reorg_batch_size): use the default value. The recommended maximum value is `1024`.

> 3. After upgrading to v7.4.0, you can adjust the number of nodes that perform backend tasks according to actual needs. After deploying the TiDB server, set the instance-level system variable [`tidb_service_scope`](/system-variables.md#tidb_service_scope-new-in-v740). Set `tidb_service_scope` to `background` to execute backend tasks. When `tidb_service_scope` is set to the default value ``, the TiDB node can not execute backend tasks. If all nodes in the cluster do not set their own `tidb_service_scope`, TiDB distributed execution framework will schedule all TiDB servers to execute backend tasks by default.

Reviewer: ```suggestion

3. After upgrading to v7.4.0, you can adjust the number of nodes that perform backend tasks according to actual needs. After deploying the TiDB server, set the instance-level system variable [`tidb_service_scope`](/system-variables.md#tidb_service_scope-new-in-v740). Set `tidb_service_scope` to `background` to execute backend tasks. When `tidb_service_scope` is set to the default value ``, the TiDB node can not execute backend tasks. If all nodes in the cluster do not set their own `tidb_service_scope`, TiDB distributed execution framework will schedule all TiDB servers to execute backend tasks by default.
```

    SET GLOBAL tidb_enable_dist_task = ON;
    ```

>     When backend tasks are running, the statements supported by the framework(Add index and `IMPORT INTO`) are executed in a distributed manner. All TiDB nodes will run backend tasks by default.

Reviewer: ```suggestion
    When backend tasks are running, the statements supported by the framework ( such as Add index and `IMPORT INTO`) are executed in a distributed manner. All TiDB nodes will run backend tasks by default.
```

2. Adjust the following system variables that might affect the distributed execution of DDL tasks according to your needs:

    * [`tidb_ddl_reorg_worker_cnt`](/system-variables.md#tidb_ddl_reorg_worker_cnt): use the default value `4`. The recommended maximum value is `16`.
    * [`tidb_ddl_reorg_priority`](/system-variables.md#tidb_ddl_reorg_priority)
    * [`tidb_ddl_error_count_limit`](/system-variables.md#tidb_ddl_error_count_limit)
    * [`tidb_ddl_reorg_batch_size`](/system-variables.md#tidb_ddl_reorg_batch_size): use the default value. The recommended maximum value is `1024`.

> 3. After upgrading to v7.4.0, you can adjust the number of nodes that perform backend tasks according to actual needs. After deploying the TiDB server, set the instance-level system variable [`tidb_service_scope`](/system-variables.md#tidb_service_scope-new-in-v740). Set `tidb_service_scope` to `background` to execute backend tasks. When `tidb_service_scope` is set to the default value ``, the TiDB node can't execute backend tasks. If all nodes in the cluster don't set their own `tidb_service_scope`, TiDB distributed execution framework will schedule all TiDB servers to execute backend tasks by default.
>     > **Warning:**
>     >
>     > This feature is an experimental feature. It is not recommended to use it in production environments.

Reviewer: ```suggestion
3. After upgrading to v7.4.0, you can adjust the number of nodes that perform backend tasks according to actual needs. After deploying the TiDB server, set the instance-level system variable [`tidb_service_scope`](/system-variables.md#tidb_service_scope-new-in-v740). Set `tidb_service_scope` to `background` to execute backend tasks. When `tidb_service_scope` is set to the default value ``, the TiDB node can not execute backend tasks. If all nodes in the cluster do not set their own `tidb_service_scope`, TiDB distributed execution framework will schedule all TiDB servers to execute backend tasks by default.

    > **Warning:**
    >
    > This feature is an experimental feature. It is not recommended to use it in production environments.

```

    SET GLOBAL tidb_enable_dist_task = ON;
    ```

>     <CustomContent platform="tidb">
> 
>     When backend tasks are running, the statements supported by the framework (such as [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) and [`IMPORT INTO`](/sql-statements/sql-statement-import-into.md)) are executed in a distributed manner. All TiDB nodes will run backend tasks by default.
> 
>     </CustomContent>
> 
>     <CustomContent platform="tidb-cloud">
> 
>     When backend tasks are running, the statements supported by the framework (such as [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) and [`IMPORT INTO`](https://docs.pingcap.com/tidb/dev/sql-statement-import-into)) are executed in a distributed manner. All TiDB nodes will run backend tasks by default.

Reviewer: ```suggestion
    <CustomContent platform="tidb">

    When background tasks are running, the statements supported by the framework (such as [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) and [`IMPORT INTO`](/sql-statements/sql-statement-import-into.md)) are executed in a distributed manner. All TiDB nodes will run background tasks by default.

    </CustomContent>

    <CustomContent platform="tidb-cloud">

    When background tasks are running, the statements supported by the framework (such as [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) and [`IMPORT INTO`](https://docs.pingcap.com/tidb/dev/sql-statement-import-into)) are executed in a distributed manner. All TiDB nodes will run background tasks by default.
```

>     When background tasks are running, the statements supported by the framework (such as [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) and [`IMPORT INTO`](/sql-statements/sql-statement-import-into.md)) are executed in a distributed manner. All TiDB nodes will run background tasks by default.
> 
>     </CustomContent>
> 
>     <CustomContent platform="tidb-cloud">
> 
>     When background tasks are running, the statements supported by the framework (such as [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) and [`IMPORT INTO`](https://docs.pingcap.com/tidb/dev/sql-statement-import-into)) are executed in a distributed manner. All TiDB nodes will run background tasks by default.

Reviewer: ```suggestion
    When background tasks are running, the statements supported by the framework (such as [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) and [`IMPORT INTO`](/sql-statements/sql-statement-import-into.md)) are executed in a distributed manner. All TiDB nodes run background tasks by default.

    </CustomContent>

    <CustomContent platform="tidb-cloud">

    When background tasks are running, the statements supported by the framework (such as [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) and [`IMPORT INTO`](https://docs.pingcap.com/tidb/dev/sql-statement-import-into)) are executed in a distributed manner. All TiDB nodes run background tasks by default.
```

>     When backend tasks are running, the statements supported by the framework (such as such as [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) and [`IMPORT INTO`](/sql-statements/sql-statement-import-into.md)) are executed in a distributed manner. All TiDB nodes will run backend tasks by default.
> 
>     </CustomContent>
> 
>     <CustomContent platform="tidb-cloud">
> 
>     When backend tasks are running, the statements supported by the framework (such as such as [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) and [`IMPORT INTO`](https://docs.pingcap.com/tidb/dev/sql-statement-import-into)) are executed in a distributed manner. All TiDB nodes will run backend tasks by default.
> 
>     </CustomContent>

Reviewer: ```suggestion
    When backend tasks are running, the statements supported by the framework (such as [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) and [`IMPORT INTO`](/sql-statements/sql-statement-import-into.md)) are executed in a distributed manner. All TiDB nodes will run backend tasks by default.

    </CustomContent>

    <CustomContent platform="tidb-cloud">

    When backend tasks are running, the statements supported by the framework (such as [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) and [`IMPORT INTO`](https://docs.pingcap.com/tidb/dev/sql-statement-import-into)) are executed in a distributed manner. All TiDB nodes will run backend tasks by default.

    </CustomContent>
```

    SET GLOBAL tidb_enable_dist_task = ON;
    ```

>     When backend tasks are running, the statements supported by the framework ( such as Add index and `IMPORT INTO`) are executed in a distributed manner. All TiDB nodes will run backend tasks by default.

Reviewer: ```suggestion
    When backend tasks are running, the statements supported by the framework (such as such as [`ADD INDEX`](/sql-statements/sql-statement-add-index.md) and [`IMPORT INTO`](/sql-statements/sql-statement-import-into.md) are executed in a distributed manner. All TiDB nodes will run backend tasks by default.
```

    * [`tidb_ddl_error_count_limit`](/system-variables.md#tidb_ddl_error_count_limit)
    * [`tidb_ddl_reorg_batch_size`](/system-variables.md#tidb_ddl_reorg_batch_size): use the default value. The recommended maximum value is `1024`.

> 3. Starting from v7.4.0, you can adjust the number of TiDB nodes that perform background tasks according to actual needs. After deploying the TiDB node, you can set the instance-level system variable [`tidb_service_scope`](/system-variables.md#tidb_service_scope-new-in-v740) to `background` to execute background tasks. When `tidb_service_scope` is set to the default value ``, the TiDB node cannot execute background tasks. If `tidb_service_scope` of all TiDB nodes in the cluster are not set, the TiDB distributed execution framework schedules all TiDB nodes to execute background tasks by default.

Reviewer 2: ```suggestion
3. Starting from v7.4.0, you can adjust the number of TiDB nodes that perform background tasks according to actual needs. After deploying a TiDB cluster, you can set the instance-level system variable [`tidb_service_scope`](/system-variables.md#tidb_service_scope-new-in-v740) of a TiDB node to `background` to execute background tasks. When `tidb_service_scope` of a TiDB node is set to the default value "", the TiDB node cannot execute background tasks. If `tidb_service_scope` is not set for any TiDB node in a cluster, the TiDB distributed execution framework schedules all TiDB nodes to execute background tasks by default.
```

> 3. After upgrading to v7.4.0, you can adjust the number of TiDB nodes that perform background tasks according to actual needs. After deploying the TiDB node, you can set the instance-level system variable [`tidb_service_scope`](/system-variables.md#tidb_service_scope-new-in-v740) to `background` to execute background tasks. When `tidb_service_scope` is set to the default value ``, the TiDB node cannot execute background tasks. If `tidb_service_scope` of all TiDB nodes in the cluster are not set, the TiDB distributed execution framework schedules all TiDB nodes to execute backend tasks by default.

Reviewer: ```suggestion
3. After upgrading to v7.4.0, you can adjust the number of TiDB nodes that perform background tasks according to actual needs. After deploying the TiDB node, you can set the instance-level system variable [`tidb_service_scope`](/system-variables.md#tidb_service_scope-new-in-v740) to `background` to execute background tasks. When `tidb_service_scope` is set to the default value ``, the TiDB node cannot execute background tasks. If `tidb_service_scope` of all TiDB nodes in the cluster are not set, the TiDB distributed execution framework schedules all TiDB nodes to execute background tasks by default.
```

> 3. After upgrading to v7.4.0, you can adjust the number of nodes that perform backend tasks according to actual needs. After deploying the TiDB server, set the instance-level system variable [`tidb_service_scope`](/system-variables.md#tidb_service_scope-new-in-v740). Set `tidb_service_scope` to `background` to execute backend tasks. When `tidb_service_scope` is set to the default value ``, the TiDB node can not execute backend tasks. If all nodes in the cluster do not set their own `tidb_service_scope`, TiDB distributed execution framework will schedule all TiDB nodes to execute backend tasks by default.

Reviewer: ```suggestion
3. After upgrading to v7.4.0, you can adjust the number of nodes that perform backend tasks according to actual needs. After deploying the TiDB node, you can set the node-level system variable [`tidb_service_scope`](/system-variables.md#tidb_service_scope-new-in-v740). Set `tidb_service_scope` to `background` to execute backend tasks. When `tidb_service_scope` is set to the default value ``, the TiDB node can not execute backend tasks. If all nodes in the cluster do not set their own `tidb_service_scope`, TiDB distributed execution framework will schedule all TiDB nodes to execute backend tasks by default.
```

> 3. After upgrading to v7.4.0, you can adjust the number of nodes that perform backend tasks according to actual needs. After deploying the TiDB server, set the instance-level system variable [`tidb_service_scope`](/system-variables.md#tidb_service_scope-new-in-v740). Set `tidb_service_scope` to `background` to execute backend tasks. When `tidb_service_scope` is set to the default value ``, the TiDB node can not execute backend tasks. If all nodes in the cluster do not set their own `tidb_service_scope`, TiDB distributed execution framework will schedule all TiDB servers to execute backend tasks by default.

Author: ```suggestion
3. After upgrading to v7.4.0, you can adjust the number of nodes that perform backend tasks according to actual needs. After deploying the TiDB server, set the instance-level system variable [`tidb_service_scope`](/system-variables.md#tidb_service_scope-new-in-v740). Set `tidb_service_scope` to `background` to execute backend tasks. When `tidb_service_scope` is set to the default value ``, the TiDB node can not execute backend tasks. If all nodes in the cluster do not set their own `tidb_service_scope`, TiDB distributed execution framework will schedule all TiDB nodes to execute backend tasks by default.
```
